lib/blob-store/index.js

```diff
@@ -89,34 +100,70 @@ export class BlobStore {
    * @param {AbortSignal} [options.signal] Optional AbortSignal to cancel in-progress download
    * @returns EventEmitter with `.state` propery, emits `state` with new state when it updates
    */
-  download (filter, { signal } = {}) {
+  download(filter, { signal } = {}) {
     return new LiveDownload(this.#hyperdrives.values(), this.#driveEmitter, {
       filter,
-      signal
+      signal,
     })
   }
 
   /**
    * @param {BlobId} blobId
    */
-  createReadStream ({ type, variant, name, driveId }) {
-    // TODO: Support 'wait' and 'timeout' opts (requires patch to Hyperdrive, or custom implementation of createReadStream)
-    const drive = this.#hyperdrives.get(driveId)
-    // TODO: Should this be an emit error on the returned stream?
-    if (!drive) throw new Error('Drive not found ' + driveId.slice(0, 7))
+  createReadStream({ type, variant, name, driveId }) {
+    // TODO: Error thrown from this be an emit error on the returned stream?
+    const drive = this.getDrive(driveId)
     const path = makePath({ type, variant, name })
     return drive.createReadStream(path)
   }
 
+  /**
+   * Optimization for creating the blobs read stream when you have
+   * previously read the entry from Hyperdrive using `drive.entry`
+   * @param {BlobId['driveId']} driveId Hyperdrive drive id
+   * @param {import('hyperdrive').HyperdriveEntry} entry Hyperdrive entry
+   */
+  async createEntryReadStream(driveId, entry) {
+    const drive = this.getDrive(driveId)
+    const blobs = await drive.getBlobs()
+
+    if (!blobs)
+      throw new Error(
+        'Hyperblobs instance not found for drive ' + driveId.slice(0, 7)
+      )
+
+    return blobs.createReadStream(entry.value.blob)
+  }
+
+  /**
+   * @param {BlobId['driveId']} driveId Hyperdrive drive id
+   * @param {import('hyperdrive').HyperdriveEntry} entry Hyperdrive entry
+   * @param {object} [opts]
+   * @param {number} [opts.length]
+   *
+   * @returns {Promise<Buffer | null>}
+   */
+  async getEntryBlob(driveId, entry, { length } = {}) {
+    const drive = this.getDrive(driveId)
+    const blobs = await drive.getBlobs()
+
+    if (!blobs)
+      throw new Error(
+        'Hyperblobs instance not found for drive ' + driveId.slice(0, 7)
+      )
+
+    return blobs.get(entry.value.blob, { wait: false, start: 0, length })
+  }
+
   /**
    *
    * @param {Omit<BlobId, 'driveId'>} blobId
    * @param {Buffer} blob
    * @param {object} [options]
-   * @param {any} [options.metadata] Metadata to store with the blob
+   * @param {{[key: string]: any, mime: string}} [options.metadata] Metadata to store with the blob
```

Comment:
okay with using `mime` as the metadata field that we store in the drive entry?

Reply:
Switched to mimeType, personal pref